gateleen-core/src/main/java/org/swisspush/gateleen/core/configuration/ConfigurationResourceValidator.java

```diff
@@ -0,0 +1,149 @@
+                return;
+            }
+
+            JsonObject schemaObject;
+            try {
+                schemaObject = new JsonObject(resourceSchema);
+            } catch(DecodeException ex) {
+                String message = "Unable to parse json schema";
+                log.warn(message);
+                future.complete(new ValidationResult(ValidationStatus.VALIDATED_NEGATIV, message));
+                return;
+            }
+
+            if(SCHEMA_DECLARATION.equals(schemaObject.getString("$schema"))) {
+                JsonSchema schema;
+                try {
+                    schema = factory.getJsonSchema(JsonLoader.fromString(resourceSchema));
+                } catch (ProcessingException | IOException e) {
+                    String message = "Cannot load schema";
+                    log.warn(message, e);
+                    future.complete(new ValidationResult(ValidationStatus.VALIDATED_NEGATIV, message));
+                    return;
+                }
+                try {
+                    ProcessingReport report = schema.validateUnchecked(JsonLoader.fromString(configurationResource.toString()));
+                    if(report.isSuccess()) {
+                        log.info("validated positive");
+                        future.complete(new ValidationResult(ValidationStatus.VALIDATED_POSITIV));
+                    } else {
+                        JsonArray validationDetails = extractMessagesAsJson(report);
+                        for(ProcessingMessage message: report) {
+                            log.warn(message.getMessage());
+                        }
+                        future.complete(new ValidationResult(ValidationStatus.VALIDATED_NEGATIV, "Validation failed", validationDetails));
+                    }
+                } catch (IOException e) {
+                    String message = "Cannot read JSON";
+                    log.warn(message, e.getMessage());
+                    future.complete(new ValidationResult(ValidationStatus.VALIDATED_NEGATIV, message));
+                }
+            } else {
+                String message = "Invalid schema: Expected property '$schema' with content '"+SCHEMA_DECLARATION+"'";
+                log.warn(message);
+                future.complete(new ValidationResult(ValidationStatus.VALIDATED_NEGATIV, message));
+            }
+        }, resultHandler);
+    }
+
+    private JsonArray extractMessagesAsJson(ProcessingReport report){
+        JsonArray resultArray = new JsonArray();
+        Iterator it = report.iterator();
+        while (it.hasNext()) {
+            ProcessingMessage msg = (ProcessingMessage) it.next();
+            JsonNode node = msg.asJson();
+            resultArray.add(new JsonObject(node.toString()));
+        }
+        return resultArray;
+    }
+
+    enum ValidationStatus {
+        VALIDATED_POSITIV,VALIDATED_NEGATIV,COULD_NOT_VALIDATE
+    }
+
+    class ValidationResult {
```

Comment:
ValidationResultalready exists in org.swisspush.gateleen.validation, shouldn't we rather move this class to gateleen-core.